presto-main/src/main/java/com/facebook/presto/SystemSessionProperties.java

```diff
@@ -283,6 +284,11 @@ public SystemSessionProperties(
                         featuresConfig.isLegacyOrderBy(),
                         false),
                 booleanSessionProperty(
+                        SUPPORT_NULL_ON_LHS_OF_SEMI_JOIN,
```

Comment:
> What about peak memory usage?

I didn't measure that. 

> AFAIR predicates extracted from EffectivePredicateExtractor are only used for equality inference. Equality inference determines if it is possible to replace one symbol with some equivalence. [...]

I see the point. Sort of. But I'm not sure this is the whole picture.

For a condition `x = c` `EffectivePredicateExtractor` states `x = c` and we know this _holds for every `x`_ regardless of `c`. (If `c` is null we know `x = c` _holds for **every** `x`_, so there is no `x`).
I _think_ we use this knowledge when handling joins.

For a projection `c as x` we cannot state `x = c` _holds for every `x`_, but `EffectivePredicateExtractor` returns the same output.

I think both kinds of information are valuable:
- (from projection) equivalence: "you can replace `x` with `c` whenever it occurs"
- (from filter) effective predicate: "`x` is equal to `c` whenever occurs"

Perhaps in JOIN both information kinds are useful and in SEMI JOIN only the latter (effective predicate) is?